src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2097,6 +2097,17 @@ public override IOperation VisitArrayElementReference(IArrayElementReferenceOper
                 operation.Syntax, operation.Type, IsImplicit(operation));
         }
 
+        public override IOperation VisitImplicitIndexerReference(IImplicitIndexerReferenceOperation operation, int? captureIdForResult)
+        {
+            EvalStackFrame frame = PushStackFrame();
+            PushOperand(VisitRequired(operation.Instance));
+            IOperation argument = VisitRequired(operation.Argument);
+            IOperation instance = PopOperand();
+            PopStackFrame(frame);
+            return new ImplicitIndexerReferenceOperation(instance, argument, operation.LengthSymbol, operation.IndexerSymbol, semanticModel: null,
```

Comment:
Should we be lowering the indexer to its final form here? I would have somewhat expected it to be turned into the series of method/property calls that make up an indexer.

Reply:
> Should we be lowering the indexer to its final form here? I would have somewhat expected it to be turned into the series of method/property calls that make up an indexer.

There is no requirement to lower things in CFG. We can accurately reflect the control flow by using the original node.